source/agora/node/Ledger.d

```diff
@@ -1236,31 +1244,54 @@ public class Ledger
     ***************************************************************************/
 
     private string isInvalidPreimageRootReason (in Height height,
-        in uint[] missing_validators) @safe nothrow
+        in uint[] missing_validators, in uint[] missing_validators_higher_bound) @safe nothrow
     {
+        import std.algorithm.setops : setDifference;
+
         Hash[] keys;
         if (!this.enroll_man.getEnrolledUTXOs(height, keys) || keys.length == 0)
             assert(0, "Could not retrieve enrollments / no enrollments found");
 
-        uint[] local_missing_validators;
+        uint[] missing_validators_lower_bound;
         foreach (idx, key; keys)
         {
             if (!this.hasRevealedPreimage(height, key))
-                local_missing_validators ~= cast(uint)idx;
+                missing_validators_lower_bound ~= cast(uint)idx;
         }
 
-        if (local_missing_validators != missing_validators)
-            return "The list of missing validators does not match with the local one. " ~
-                assumeWontThrow(to!string(missing_validators)) ~
-                " != " ~ assumeWontThrow(to!string(local_missing_validators));
+        // The idea here is that, the previous way of implementing what is
+        // considered 'acceptable candidate regarding missing validators'
+        // was too strict. Instead of equality, NodeA will check the candidate
+        // from NodeB in the following way:
```

Comment:
Context fades, and any documentation relying on too much context will eventually become problematic.
Which is why documenting a functionality based on a (assumedly known) previous state is not future-proof.
If you want to mention the previous state, do so explicitly:
> A previous implementation that was simply checking that the local node's missing validators
> was equal to the one received was inadequate because [...]

Or simply document the state of things as it is now:
> NodeA will check the candidate [...]

I do think it's good to mention why a simple equality check is not a good idea, but the rationale should be provided.

Reply:
fixed the doc